### ISSUE TITLE
internal/modules/cjs/loader.js 584 throw err; ==> cannot find module core-util-is

### ISSUE THREAD

**Issue by subhrajitroy1989** [nodejs/help#1846]
* **Version**:
* **Platform**:
* **Subsystem**:

**Comment by yathamravali** [comment]
@subhrajitroy1989 
Try
`rm -rf node_modules`
`npm install -g npm@latest`
`npm i core-util-is`

**Comment by Sruth8** [comment]
Does anyone know if this works. I'm getting the same problem.

**Comment by Sruth8** [comment]
I just did the steps above and I'm still getting the error. What should I do next.

**Comment by yathamravali** [comment]
@Sruth8 
What's the node and npm version you are using?
Please post the package-json

**Comment by LeonelVinasco** [comment]
@Sruth8 It worked for me.

**Comment by jluisborjon** [comment]
It worked! Thanks

**Comment by zerefdragneel3007** [comment]
getting same error when i run express server through node main.js

package.json
{
  "name": "app",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "start:web": "node web/dist/bundle.js",
    "test": "jest",
    "build:dev": "rm -rf web/dist web/public && webpack --mode=development",
    "build:pro": "rm -rf web/dist web/public && webpack --mode=production",
    "serve:pro": "rm -rf web/dist web/public && webpack-dev-server --mode=production --hot",
    "serve:dev": "rm -rf web/dist web/public && webpack-dev-server --mode=development --hot",
    "desktop": "electron ."
  },
  "dependencies": {
    "react": "16.8.3",
    "react-dom": "^16.8.6",
    "react-native": "^0.59.8",
    "react-native-web": "^0.11.2"
  },
  "devDependencies": {
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.4.4",
    "babel-jest": "^24.8.0",
    "babel-loader": "^8.0.5",
    "css-loader": "^2.1.1",
    "express": "^4.16.4",
    "extract-text-webpack-plugin": "^4.0.0-beta.0",
    "file-loader": "^3.0.1",
    "html-webpack-plugin": "^3.2.0",
    "jest": "^24.8.0",
    "metro-react-native-babel-preset": "^0.54.0",
    "react-test-renderer": "16.8.3",
    "style-loader": "^0.23.1",
    "webpack": "^4.31.0",
    "webpack-cli": "^3.3.2",
    "webpack-dev-server": "^3.3.1",
    "webpack-node-externals": "^1.7.2"
  },
  "jest": {
    "preset": "react-native"
  }
}

error=>
> node web/dist/bundle.js

internal/modules/cjs/loader.js:584
    throw err;
    ^

Error: Cannot find module 'warnOnce'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/home/ravi/Documents/practice/node/react/Cross-Platefrorm (copy)/app/node_modules/react-native/Libraries/react-native/react-native-implementation.js:14:18)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! app@0.0.1 start:web: `node web/dist/bundle.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the app@0.0.1 start:web script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ravi/.npm/_logs/2019-05-18T17_15_13_253Z-debug.log

**Comment by Amirm73** [comment]
use this method to remove npm and node:
https://stackoverflow.com/questions/20711240/how-to-completely-remove-node-js-from-windows

then install nodejs LTS from nodejs.com

**Comment by prasannaballal** [comment]
> @subhrajitroy1989
> `npm i core-util-is`

doing this only worked for me.

**Comment by gireeshpunathil** [comment]
@subhrajitroy1989 - is this resolved?

**Comment by radurentea** [comment]
What worked for me is to make sure the webpack-dev-server is up to date, so run the following command:
`npm install webpack-dev-server`

**Comment by gireeshpunathil** [comment]
resolved, closing

**Comment by tarunmanchukonda** [comment]
Tried each and everything mentioned here but nothing is working out.

### npm ERR! path /home/tarunmanchukonda/package.json
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall open
npm ERR! enoent ENOENT: no such file or directory, open '/home/tarunmanchukonda/package.json'
npm ERR! enoent This is related to npm not being able to find a file.



### npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tarunmanchukonda/.npm/_logs/2019-07-19T12_54_53_629Z-debug.log

**Comment by maakireddymaheswarReddy** [comment]
> Does anyone know if this works. I'm getting the same problem.

body-parser type like this then it works

**Comment by rspixers** [comment]
I'm getting an error like this

internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module './paths'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\Users\Prashanth\Desktop\React\react-boilerplategit\config\webpack/webpack.common.babel.js:4:1)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Module._compile (C:\Users\Prashanth\Desktop\React\react-boilerplategit\node_modules\pirates\lib\index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Object.newLoader [as .js] (C:\Users\Prashanth\Desktop\React\react-boilerplategit\node_modules\pirates\lib\index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)

These are my Dev Dependencies:
"devDependencies": {
    "@babel/cli": "7.4.3",
    "@babel/core": "7.4.3",
    "@babel/node": "7.2.2",
    "@babel/plugin-proposal-class-properties": "7.3.0",
    "@babel/plugin-proposal-object-rest-spread": "7.3.1",
    "@babel/plugin-syntax-dynamic-import": "7.2.0",
    "@babel/plugin-syntax-import-meta": "7.2.0",
    "@babel/plugin-transform-async-to-generator": "7.2.0",
    "@babel/plugin-transform-runtime": "7.2.0",
    "@babel/preset-env": "7.3.1",
    "@babel/preset-react": "7.0.0",
    "@babel/register": "7.0.0",
    "@babel/runtime": "7.3.1",
    "babel-core": "7.0.0-bridge.0",
    "babel-eslint": "10.0.1",
    "babel-jest": "24.0.0",
    "babel-loader": "8.0.4",
    "clean-webpack-plugin": "^1.0.1",
    "core-js": "3",
    "css-loader": "^2.1.0",
    "cypress": "^3.1.4",
    "eslint": "^5.12.1",
    "eslint-config-airbnb": "^17.1.0",
    "eslint-config-prettier": "^4.0.0",
    "eslint-loader": "^2.1.1",
    "eslint-plugin-babel": "^5.3.0",
    "eslint-plugin-import": "^2.15.0",
    "eslint-plugin-jsx-a11y": "^6.2.0",
    "eslint-plugin-prettier": "^3.0.0",
    "eslint-plugin-react": "^7.12.4",
    "extract-text-webpack-plugin": "^3.0.2",
    "file-loader": "^3.0.1",
    "html-webpack-plugin": "^3.2.0",
    "husky": "^1.2.0",
    "img-loader": "^3.0.1",
    "jest": "^24.0.0",
    "jest-dom": "^3.0.1",
    "node-sass": "^4.10.0",
    "postcss-loader": "^3.0.0",
    "prettier": "^1.16.1",
    "react-testing-library": "^7.0.1",
    "redux-logger": "^3.0.6",
    "sass-loader": "^7.1.0",
    "style-loader": "^0.23.1",
    "url-loader": "^1.1.2",
    "webpack": "^4.32.0",
    "webpack-cli": "^3.3.4",
    "webpack-dev-server": "^3.7.2",
    "webpack-jarvis": "^0.3.2",
    "webpack-merge": "^4.2.1"
  },
  "dependencies": {
    "core-util-is": "^1.0.2",
    "cross-env": "^5.2.0",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-redux": "^7.1.0",
    "react-router-dom": "^5.0.1",
    "redux": "^4.0.1",
    "redux-thunk": "^2.3.0",
    "styled-components": "^4.3.1"
  }

**Comment by abu-bakarr** [comment]
i need help 


[nodemon] 1.18.9
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node ejs, js, json app`
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'bcrypt'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\Users\Lovel\Desktop\Greenta-Collect-App\app\routes\adminRoutes.js:4:16)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\Users\Lovel\Desktop\Greenta-Collect-App\app.js:152:21)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
[nodemon] app crashed - waiting for file changes before starting...

**Comment by JakeGreer** [comment]
Struggled with this error all day using a MERN stack. I fixed it by adding this line to my package.json in on the server side.

"client": "npm start --prefix client"

**Comment by zerefdragneel3007** [comment]
> I'm getting an error like this
> 
> internal/modules/cjs/loader.js:638
> throw err;
> ^
> 
> Error: Cannot find module './paths'
> at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
> at Function.Module._load (internal/modules/cjs/loader.js:562:25)
> at Module.require (internal/modules/cjs/loader.js:690:17)
> at require (internal/modules/cjs/helpers.js:25:18)
> at Object. (C:\Users\Prashanth\Desktop\React\react-boilerplategit\config\webpack/webpack.common.babel.js:4:1)
> at Module._compile (internal/modules/cjs/loader.js:776:30)
> at Module._compile (C:\Users\Prashanth\Desktop\React\react-boilerplategit\node_modules\pirates\lib\index.js:99:24)
> at Module._extensions..js (internal/modules/cjs/loader.js:787:10)
> at Object.newLoader [as .js] (C:\Users\Prashanth\Desktop\React\react-boilerplategit\node_modules\pirates\lib\index.js:104:7)
> at Module.load (internal/modules/cjs/loader.js:653:32)
> 
> These are my Dev Dependencies:
> "devDependencies": {
> "@babel/cli": "7.4.3",
> "@babel/core": "7.4.3",
> "@babel/node": "7.2.2",
> "@babel/plugin-proposal-class-properties": "7.3.0",
> "@babel/plugin-proposal-object-rest-spread": "7.3.1",
> "@babel/plugin-syntax-dynamic-import": "7.2.0",
> "@babel/plugin-syntax-import-meta": "7.2.0",
> "@babel/plugin-transform-async-to-generator": "7.2.0",
> "@babel/plugin-transform-runtime": "7.2.0",
> "@babel/preset-env": "7.3.1",
> "@babel/preset-react": "7.0.0",
> "@babel/register": "7.0.0",
> "@babel/runtime": "7.3.1",
> "babel-core": "7.0.0-bridge.0",
> "babel-eslint": "10.0.1",
> "babel-jest": "24.0.0",
> "babel-loader": "8.0.4",
> "clean-webpack-plugin": "^1.0.1",
> "core-js": "3",
> "css-loader": "^2.1.0",
> "cypress": "^3.1.4",
> "eslint": "^5.12.1",
> "eslint-config-airbnb": "^17.1.0",
> "eslint-config-prettier": "^4.0.0",
> "eslint-loader": "^2.1.1",
> "eslint-plugin-babel": "^5.3.0",
> "eslint-plugin-import": "^2.15.0",
> "eslint-plugin-jsx-a11y": "^6.2.0",
> "eslint-plugin-prettier": "^3.0.0",
> "eslint-plugin-react": "^7.12.4",
> "extract-text-webpack-plugin": "^3.0.2",
> "file-loader": "^3.0.1",
> "html-webpack-plugin": "^3.2.0",
> "husky": "^1.2.0",
> "img-loader": "^3.0.1",
> "jest": "^24.0.0",
> "jest-dom": "^3.0.1",
> "node-sass": "^4.10.0",
> "postcss-loader": "^3.0.0",
> "prettier": "^1.16.1",
> "react-testing-library": "^7.0.1",
> "redux-logger": "^3.0.6",
> "sass-loader": "^7.1.0",
> "style-loader": "^0.23.1",
> "url-loader": "^1.1.2",
> "webpack": "^4.32.0",
> "webpack-cli": "^3.3.4",
> "webpack-dev-server": "^3.7.2",
> "webpack-jarvis": "^0.3.2",
> "webpack-merge": "^4.2.1"
> },
> "dependencies": {
> "core-util-is": "^1.0.2",
> "cross-env": "^5.2.0",
> "react": "^16.8.6",
> "react-dom": "^16.8.6",
> "react-redux": "^7.1.0",
> "react-router-dom": "^5.0.1",
> "redux": "^4.0.1",
> "redux-thunk": "^2.3.0",
> "styled-components": "^4.3.1"
> }

try removeing style-loader, that helped me with my server side rendering

**Comment by 21511012** [comment]
llllllllllll

**Comment by yourwpmadesimple** [comment]
@subhrajitroy1989 ,  My brother from a different mother. Thank you, man!

**Comment by qureshitauqeer3** [comment]
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'C:\Users\admin'\ScrapeExample.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)

**Comment by shanmicheal25** [comment]
{
	"name": "app",
	"main": "server.js",
	"dependencies": {
		"bcrypt": "^3.0.6",
		"body-parser": "^1.19.0",
		"cron-parser": "^2.10.0",
		"cronstrue": "^1.65.0",
		"dateformat": "^3.0.3",
		"dotenv": "^7.0.0",
		"express": "^4.17.0",
		"fs": "0.0.1-security",
		"http": "0.0.0",
		"https": "^1.0.0",
		"jsonwebtoken": "^8.5.1",
		"lokijs": "^1.5.6",
		"mariadb": "^2.0.5",
		"node-cron": "^2.0.3",
		"request": "^2.88.0",
		"sftp-upload": "^1.0.1",
		"ssh2-sftp-client": "^2.5.0",
		"ssl-root-cas": "^1.3.1"
	},
	"scripts": {
		"start": "nodemon --ignore public/data --ignore db -r dotenv/config server.js"
	},
	"devDependencies": {
		"nodemon": "^1.19.1"
	}
}

**Comment by shanmicheal25** [comment]
I facing the same problem.

**Comment by hubcelab** [comment]
All you need to do it install ng using this command

npm install -g @angular/cli

www.hubcelab.com

**Comment by hubcelab** [comment]
All you need to do is 

**npm install -g @angular/cli**

https://www.hubcelab.com

**Comment by OlatunjiGab** [comment]
ionic-app-scripts.cmd build --target cordova --platform android
[20:28:37]  ionic-app-scripts 3.2.4 
[20:28:38]  build dev started ...
[20:28:39]  clean started ...
[20:28:42]  clean finished in 2.80 s
[20:28:42]  copy started ...
[20:28:52]  deeplinks started ...
[20:28:52]  deeplinks finished in 895 ms
[20:28:53]  transpile started ...
[20:30:06]  transpile finished in 73.12 s
[20:30:06]  preprocess started ...
[20:30:06]  preprocess finished in 343 ms
[20:30:07]  webpack started ...
[20:32:48]  webpack finished in 161.67 s
[20:32:49]  sass started ...
[20:34:00]  sass finished in 71.63 s
[20:35:13]  copy finished in 391.25 s
[20:35:14]  postprocess started ...
[20:35:17]  postprocess finished in 3.64 s
[20:35:18]  lint started ...
[20:35:25]  build dev finished in 407.58 s
[20:38:48]  lint finished in 210.07 s
> cordova.cmd build android
internal/modules/cjs/loader.js:584
    throw err;
    ^

Error: Cannot find module 'update-notifier'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (C:\Users\Gabriel\Documents\GabFile\MyProject\Book-Store\node_modules\cordova\src\cli.js:20:22)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
[ERROR] An error occurred while running subprocess cordova.

        cordova.cmd build android exited with exit code 1.

        Re-running this command with the --verbose flag may provide more information.
This error has been delaying my job any help pls, It works fine on the browser but any time i want to deploy on device it throws this error

**Comment by Harizriz** [comment]
I want to run my **Vue.js** using **npm** on localhost:8080 but they give me this error. Is there a way to solve this issue?

This error only occurs when I tried to install **node_modules** and **package-lock.json** in my Vue folder containing **.babelrc**, **package.json** and **webpack.config.js**. For my other Vue folders, I still can run my localhost:8080 by using the same method for all folders which is

```
npm install
npm run dev
```

The code below is the Error.

```
The system cannot find the path specified.
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'C:\Users\acer\Documents\VueJs\cross-env\dist\bin\cross-env.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! vue-cli@ dev: `cross-env NODE_ENV=development webpack-dev-server --open --inline --hot`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the vue-cli@ dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\acer\AppData\Roaming\npm-cache\_logs\2019-08-26T02_18_38_374Z-debug.log
```

**Comment by twilsonpierce** [comment]
> @subhrajitroy1989
> Try
> `rm -rf node_modules`
> `npm install -g npm@latest`
> `npm i core-util-is`

Thanks this worked for me!

**Comment by Achala03** [comment]
Getting error .Working on Laravel-mix and react

E:\startupwala-website>npm run watch -poll

> startupwala-website@1.0.0 watch E:\startupwala-website
> npm run development -- --watch


> startupwala-website@1.0.0 development E:\startupwala-website
> cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js "--watch"

internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'webpack/lib/RuleSet'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (E:\startupwala-website\node_modules\vue-loader\lib\plugin.js:2:17)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! startupwala-website@1.0.0 development: `cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js "--watch"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the startupwala-website@1.0.0 development script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Achala\AppData\Roaming\npm-cache\_logs\2019-09-05T12_39_53_335Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! startupwala-website@1.0.0 watch: `npm run development -- --watch`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the startupwala-website@1.0.0 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Achala\AppData\Roaming\npm-cache\_logs\2019-09-05T12_39_53_398Z-debug.log

**Comment by Achala03** [comment]
Windows 10
Laravel-mix
React

I Tried below cmd but still getting errors ....

rm -rf node_modules
npm install -g npm@latest
npm i core-util-is

E:\startupwala-website>npm run watch -poll

> startupwala-website@1.0.0 watch E:\startupwala-website
> npm run development -- --watch


> startupwala-website@1.0.0 development E:\startupwala-website
> cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js "--watch"

E:\startupwala-website\webpack.mix.js:10
mix.disableNotifications();
    ^

TypeError: Cannot read property 'disableNotifications' of undefined
    at Object.<anonymous> (E:\startupwala-website\webpack.mix.js:10:5)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (E:\startupwala-website\node_modules\laravel-mix\setup\webpack.config.js:12:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! startupwala-website@1.0.0 development: `cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js "--watch"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the startupwala-website@1.0.0 development script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Achala\AppData\Roaming\npm-cache\_logs\2019-09-05T13_16_02_205Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! startupwala-website@1.0.0 watch: `npm run development -- --watch`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the startupwala-website@1.0.0 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Achala\AppData\Roaming\npm-cache\_logs\2019-09-05T13_16_02_354Z-debug.log

**Comment by chavychaze** [comment]
I had installed `@nestjs/cli` globally, so delete manually folder and reinstalling it works for me

**Comment by katrinanova** [comment]
also solved `internal/modules/cjs/loader.js throw err; ==> Cannot find module './codes.json'`

**Comment by victorperalta93** [comment]
`sudo npm i core-util-is` fixed it for me.

**Comment by ZpeedTube** [comment]
My error was:
```
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'theFilePath'
```
I think I had some other error number at some point also.

Using:
Win 10
nodejs v10.16.3
typescript, express, mongoose, (more modules that are not relevant)

**_Scenario:_** I was trying to use a file outside **src** folder. When compiling and trying to run I got that error.

**_My solution was:_** I moved the file from outside **src** in to the **src** folder and the error was gone.

Hope this helps someone doing the same thing as me. 🙂

**Comment by Zacknero** [comment]
> internal/modules/cjs/loader.js:638
> throw err;
> ^
> 
> Error: Cannot find module 'C:\Users\admin'\ScrapeExample.js'
> at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
> at Function.Module._load (internal/modules/cjs/loader.js:562:25)
> at Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
> at startup (internal/bootstrap/node.js:283:19)
> at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)

How you solved it?? I have the same problem

**Comment by QuentinIcky** [comment]
Verify your hierarchy. I already had this issue, my server.js was in the wrong directory. Everything worked after

**Comment by Zacknero** [comment]
> Verify your hierarchy. I already had this issue, my server.js was in the wrong directory. Everything worked after

server.js?? What's this file? But the scaffold is builded by nestjs-cli generator.

**Comment by QuentinIcky** [comment]
@Zacknero in your case I think ScrapeExample.js is the file corresponding to my server.js

**Comment by Zacknero** [comment]
@QuentinIcky i never created the ScrapeExample.js...this create the cli of nestjs....Because i don't find this on my pc :S :S

**Comment by Zorono** [comment]
Please some help i tried the following commands:
```
npm install -g npm@latest
npm i core-util-is
```
but it does nothing useful to me :(
```
C:\Users\BrownTurbo Gaming>node C:\Users\BrownTurboGaming\Desktop\NodeJS\CreativeGeneration\index.js
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'C:\Users\BrownTurbo'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
```

**Comment by chavychaze** [comment]
Mostly this error's occured when the nestjs wasn't installed correct globally, the best option to fix this, is try to reinstall it, in case, that after creating the file manually to fix one issue, you'll got to do it each time, when generating new modules with nest. Hope this help someone

**Comment by Zacknero** [comment]
> Mostly this error's occured when the nestjs wasn't installed correct globally, the best option to fix this, is try to reinstall it, in case, that after creating the file manually to fix one issue, you'll got to do it each time, when generating new modules with nest. Hope this help someone

Ok but i have NVM, i deleted nvm node and re-installed but don't work

**Comment by DhritishRoy** [comment]
I am also having the same problem just like @Zacknero , the same error and I tried by reinstalling npm globally also I checked my directory also and  the npm installed is also of latest version.
Can someone kindly sort out this problem, will be a real help.

**Comment by samayo** [comment]
I solved my issue by removing single quotes `'` from my `package.json` config.

ex, from
```js
"scripts": {
 "dev": "webpack-dev-server --config './webpack.config.js'",
```
to
```js
"scripts": {
 "dev": "webpack-dev-server --config ./webpack.config.js",
```
If this doesn't work, make sure your `"main": {..}` actually points to the main file, and delete `node_modules/`, `package-lock.json` and install evertything `npm i`

**Comment by Startupwala** [comment]
Thank u .It is working fine now :)..

**Comment by Zacknero** [comment]
> ackage.jso

My package.json have this scripts:

`  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "tslint -p tsconfig.json -c tslint.json",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  }`

I don't have your dev script, or must add it? Thanks!

**Comment by samayo** [comment]
Nah, dev is just an identifier to start the webpack dev server, in your case it is `start:dev` your should probably delete node_modules, package-lock.json and run `npm i`

**Comment by Zacknero** [comment]
> Nah, dev is just an identifier to start the webpack dev server, in your case it is `start:dev` your should probably delete node_modules, package-lock.json and run `npm i`

Don't work

**Comment by birender** [comment]
Hi
I am Facing Same Problem
after install vue on command prompt
npm install -g @vue/cli

Still vue not recognized by command prompt

Window 10 Home Edition

Thanks

**Comment by ocBruno** [comment]
I have repeatedly received this error from internal/modules/cjs/loader.js on my windows environment on various projects. 
I have noticed that most of these projects that raise htis error have dependencies which depend on core-js@<3.0 which is not mainted becaused of the amount of issues. Does anyone know if this is one of the recurrent issues?
I follow the same setup instructions for the projects on my ubuntu virtualbox and everything works fine so I have been trying to debug my node installation and my environment.
I tried various solutions and what is working for me is probably not advised but for now i'm tired of debugging my environment :/
What I have been doing to work on windows 10 is not installing with yarn and using npm i --unsafe-perm which probably isn't advised but considering the packages that are failing aren't required for development and everything works fine as far as I have seen for now it will do...

**Comment by TheNightbear** [comment]
Had the same problem under macOS Catalina.
A reinstall of the latest Node LTS version from (https://nodejs.org/de/download/) solved my problem.

**Comment by ymkarthik** [comment]
I have a similar problem while i am running a docker image

**Comment by ymkarthik** [comment]
I have a similar problem while i am running a docker image

**Comment by ymkarthik** [comment]
I have a similar problem while i am running a docker image

-> docker run my_hub_user_name/select
internal/modules/cjs/loader.js:800
    throw err;
    ^

Error: Cannot find module 'mysql'
Require stack:
- /usr/src/app/select.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
    at Function.Module._load (internal/modules/cjs/loader.js:690:27)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/usr/src/app/select.js:1:13)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/usr/src/app/select.js' ]
}

**Comment by shubhsk88** [comment]
I am getting the same error while deploying my app to heroku while it was working perfectly fine on my computer
```2020-01-04T18:29:12.996429+00:00 heroku[web.1]: Starting process with command `node server.js`
2020-01-04T18:29:15.402134+00:00 heroku[web.1]: Process exited with status 1
2020-01-04T18:29:15.419066+00:00 heroku[web.1]: State changed from starting to crashed
2020-01-04T18:29:15.342265+00:00 app[web.1]: internal/modules/cjs/loader.js:800
2020-01-04T18:29:15.342286+00:00 app[web.1]: throw err;
2020-01-04T18:29:15.342288+00:00 app[web.1]: ^
2020-01-04T18:29:15.342291+00:00 app[web.1]: 
2020-01-04T18:29:15.342293+00:00 app[web.1]: Error: Cannot find module 'express'
2020-01-04T18:29:15.342295+00:00 app[web.1]: Require stack:
2020-01-04T18:29:15.342298+00:00 app[web.1]: - /app/server.js
2020-01-04T18:29:15.342300+00:00 app[web.1]: at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
2020-01-04T18:29:15.342302+00:00 app[web.1]: at Function.Module._load (internal/modules/cjs/loader.js:690:27)
2020-01-04T18:29:15.342304+00:00 app[web.1]: at Module.require (internal/modules/cjs/loader.js:852:19)
2020-01-04T18:29:15.342307+00:00 app[web.1]: at require (internal/modules/cjs/helpers.js:74:18)
2020-01-04T18:29:15.342309+00:00 app[web.1]: at Object.<anonymous> (/app/server.js:1:15)
2020-01-04T18:29:15.342311+00:00 app[web.1]: at Module._compile (internal/modules/cjs/loader.js:959:30)
2020-01-04T18:29:15.342313+00:00 app[web.1]: at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
2020-01-04T18:29:15.342316+00:00 app[web.1]: at Module.load (internal/modules/cjs/loader.js:815:32)
2020-01-04T18:29:15.342318+00:00 app[web.1]: at Function.Module._load (internal/modules/cjs/loader.js:727:14)
2020-01-04T18:29:15.342320+00:00 app[web.1]: at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10) {
2020-01-04T18:29:15.342322+00:00 app[web.1]: code: 'MODULE_NOT_FOUND',
2020-01-04T18:29:15.342324+00:00 app[web.1]: requireStack: [ '/app/server.js' ]
2020-01-04T18:29:15.342327+00:00 app[web.1]: }
```

**Comment by kaaranRaahulS** [comment]
> What worked for me is to make sure the webpack-dev-server is up to date, so run the following command:
> `npm install webpack-dev-server`

This worked out well !

**Comment by subhrajitroy1989** [comment]
Hi Guys,

If you have a project dependency on a particular version of npm, please
don't update it. It's hectic to update the dependencies and resolve them
one by one. So, it would be wise to uninstall the existing core-util.js
which got corrupt. Please install the compatible core-util.js for the npm
version and the issue will get resolved.

Thanks & Regards

Subhrajit Roy

**Comment by ArunTeltia** [comment]
okay i got the same error because i was not in my project main file i was somewhere outside my file and after changing my  directory i dont get that error anymore

**Comment by shubhsk88** [comment]
I solved the error by making the changes in my package.json file and when I installed express in my localhost it was mentioned in devDependencies and when I deployed to Heroku I changed the express mention from devDependencies to dependencies and it works out very well. I hope this will be helpful.It took me so long to correct it

**Comment by mrferris** [comment]
Has anybody explained **why** this happens? I read this thread and may have missed it. I didn't change anything about my code and suddenly got this error. I'm running on an EC2, I wonder if something about the system changed? 

Regardless, `rm -rf node_modules && rm package-lock.json && npm install` fixed this for me. Thanks!

**Comment by tax5** [comment]
Error: Cannot find module 'on-finished'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/rbd/pnpm-volume/d3bdb10c-84a9-406d-8738-8a8e61d05f28/node_modules/.registry.npmjs.org/finalhandler/1.1.2/node_modules/finalhandler/index.js:17:18)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)

**Comment by misolo** [comment]
I had the same error after upgrading the node, was resolved simply by restarting the computer

**Comment by KelseyMetter** [comment]
I can't figure this out. I've been trying all day.                                                                               
                   
    === npm audit security report ===                        
                                                                                
                                                                                
                                 Manual Review                                  
             Some vulnerabilities require your attention to resolve             
                                                                                
          Visit https://go.npm.me/audit-guide for additional guidance           
                                                                                
                                                                                
  Low             Incorrect Handling of Non-Boolean Comparisons During          
                  Minification                                                  
                                                                                
  Package         uglify-js                                                     
                                                                                
  Patched in      >= 2.4.24                                                     
                                                                                
  Dependency of   jade                                                          
                                                                                
  Path            jade > transformers > uglify-js                               
                                                                                
  More info       https://npmjs.com/advisories/39                               
                                                                                
                                                                                
  Low             Regular Expression Denial of Service                          
                                                                                
  Package         uglify-js                                                     
                                                                                
  Patched in      >=2.6.0                                                       
                                                                                
  Dependency of   jade                                                          
                                                                                
  Path            jade > transformers > uglify-js                               
                                                                                
  More info       https://npmjs.com/advisories/48                               
                                                                                
                                                                                
  Critical        Sandbox Bypass Leading to Arbitrary Code Execution            
                                                                                
  Package         constantinople                                                
                                                                                
  Patched in      >=3.1.1                                                       
                                                                                
  Dependency of   jade                                                          
                                                                                
  Path            jade > constantinople                                         
                                                                                
  More info       https://npmjs.com/advisories/568                              
                                                                                
                                                                                
  Low             Regular Expression Denial of Service                          
                                                                                
  Package         clean-css                                                     
                                                                                
  Patched in      >=4.1.11                                                      
                                                                                
  Dependency of   jade                                                          
                                                                                
  Path            jade > clean-css                                              
                                                                                
  More info       https://npmjs.com/advisories/785                              
                                                                                
found 4 vulnerabilities (3 low, 1 critical) in 3752 scanned packages
  4 vulnerabilities require manual review. See the full report for details.

**Comment by prathyushaRed** [comment]
> @Sruth8
> What's the node and npm version you are using?
> Please post the package-json

o

**Comment by prathyushaRed** [comment]
where to post on cmd or

**Comment by KelseyMetter** [comment]
C:\Users\kelse>node --versionv12.16.1  Sent from Mail for Windows 10 From: prathyushaRedSent: Monday, March 2, 2020 11:08 PMTo: nodejs/helpCc: Kelsey Metter; CommentSubject: Re: [nodejs/help] internal/modules/cjs/loader.js 584 throw err; ==> cannot find module core-util-is (#1846) where to post on cmd or—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by KelseyMetter** [comment]
C:\Users\kelse>node --version
v12.16.1

**Comment by tax5** [comment]
Error: Cannot find module 'express'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
at Object.<anonymous> (/app/server.js:7:17)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
🐞 The debugger could not be started. See https://glitch.com/faq#debugger
internal/modules/cjs/loader.js:584
    throw err;
    ^
Error: Cannot find module 'express'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
at Object.<anonymous> (/app/server.js:7:17)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)

**Comment by anibabbar** [comment]
npm run clean && npm run build worked for me !!

**Comment by theslyprofessor** [comment]
> npm run clean && npm run build worked for me !!

When I try that I get npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /Users/nakultiruviluamala/package.json

I'm really desperate to get this to work.

Thanks,
Nakul

**Comment by anibabbar** [comment]
> > npm run clean && npm run build worked for me !!
> 
> When I try that I get npm ERR! code ENOENT
> npm ERR! syscall open
> npm ERR! path /Users/nakultiruviluamala/package.json
> 
> I'm really desperate to get this to work.
> 
> Thanks,
> Nakul

try : https://github.com/nodejs/help/issues/1846#issuecomment-480563345

**Comment by theslyprofessor** [comment]
> > > npm run clean && npm run build worked for me !!
> > 
> > 
> > When I try that I get npm ERR! code ENOENT
> > npm ERR! syscall open
> > npm ERR! path /Users/nakultiruviluamala/package.json
> > I'm really desperate to get this to work.
> > Thanks,
> > Nakul
> 
> try : [#1846 (comment)](https://github.com/nodejs/help/issues/1846#issuecomment-480563345)

When I do the last step:npm i core-util-is

I get this error: 
npm WARN saveError ENOENT: no such file or directory, open '/Users/nakultiruviluamala/package.json'

Any idea why?

Thanks,
Nakul

**Comment by aSrivastaava** [comment]
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'uws'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (F:/chatapp/src/index.js:7:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at loader (F:\chatapp\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (F:\chatapp\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)

**Comment by elderon89** [comment]
Could someone help?

internal/modules/cjs/loader.js:979
  throw err;
  ^

Error: Cannot find module 'E:\Projetos\testes\ubots\backend\index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:976:15)     
    at Function.Module._load (internal/modules/cjs/loader.js:859:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
[nodemon] app crashed - waiting for file changes before starting...

Did everything above, with no results :(

**Comment by AksCA** [comment]
npm install succeeded. However on running "next build src" command it fails with the following message in linux machine.

    next build src

sh: next: not found 
npm ERR! code ELIFECYCLE 
npm ERR! syscall spawn 
npm ERR! file sh 
npm ERR! errno ENOENT 
npm ERR! @fl-ukdeploy/fsheet-ui@1.0.0 build: next build src 
npm ERR! spawn ENOENT 
npm ERR! npm ERR! Failed at the @fl-ukdeploy/fsheet-ui@1.0.0 build script. npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in: npm ERR! /appuser/.npm/_logs/2020-04-10T13_12_27_971Z-debug.log

    node --version v12.12.0
    npm --version 6.11.3

Digging further found the following:

    ls -latr ./node_modules/next/dist/bin total 8 -rwxr-xr-x 1 appuser appgroup 4283 Oct 26 1985 next drwxr-xr-x 13 appuser appgroup 159 Apr 10 13:21 .. drwxr-xr-x 2 appuser appgroup 18 Apr 10 13:25 .

    ls -latr ./node_modules/.bin/next ls: ./node_modules/.bin/next: No such file or directory

This happens some times. Some times it worked as well.

Any inputs on this please?

**Comment by LuizGMX** [comment]
> npm run clean && npm run build worked for me !!

after many tries, this was the only one that worked for me Thanks!!
what i did:
npm install clean
npm clean
npm build

**Comment by faizantariq1** [comment]
**Note:** It's running on android side but giving error on ios side. when adding realm library.

Using Node.js v12.16.2
internal/modules/cjs/loader.js:983
  throw err;
  ^

Error: Cannot find module '/Users/faraztariq/Desktop/Naseeb'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
Command PhaseScriptExecution failed with a nonzero exit code

**Comment by TecateWine** [comment]
Anyone help me with this? I have uninstalled, reinstalled, and repaired nodejs probably half a dozen times now. I get this error with every npm command. OS: Win10:

internal/modules/cjs/loader.js:960
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\bin\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

**Comment by dfrank43** [comment]
I am getting the exact same thing.  What is the deal???

**Comment by faizantariq1** [comment]
@dfrank43  @TecateWine in my case.
I have a space in source repo "folder name" in my system. I just remove that space and It's work for me.

**Comment by peterl74** [comment]
> @dfrank43 @TecateWine in my case.
> I have a space in source repo "folder name" in my system. I just remove that space and It's work for me.

I  had an ampersand in the folder name, thanks.

**Comment by anguschp** [comment]
> okay i got the same error because i was not in my project main file i was somewhere outside my file and after changing my directory i dont get that error anymore so please check for that as well if you are getting that error

This works. I cloned the whole project in a directory and that's create a sub folder inside another folder. After re-arrange the diredctory, problem solved

**Comment by marcosparreira10** [comment]
I corrected this error, returning to the visual code, deleting the file and making the file again. It seems that this file only recognized in the visual code, that if it is in the project folder the file is not there. And this error is the error of not having found the file. That's how I fixed this defect. My file that reads the cmd port, server.js misses

**Comment by abdelrhmanshokr** [comment]
I was getting this error => 
Error: Cannot find module 'my_file_path/index.js'

and for some reason in package.json main field was set to index.js and my main file was called app.js  so I changed  "main": "index.js",  to  "main": "app.js",

**Comment by SVladikO** [comment]
I've got this issue because I was in wrong folder.

**Comment by SathwikaGone** [comment]
> Has anybody explained **why** this happens? I read this thread and may have missed it. I didn't change anything about my code and suddenly got this error. I'm running on an EC2, I wonder if something about the system changed?
> 
> Regardless, `rm -rf node_modules && rm package-lock.json && npm install` fixed this for me. Thanks!

hey thanks that worked for me. I just followed these three commands > rm -rf node_modules && rm package-lock.json && npm install. THANK YOU

**Comment by ridwan102** [comment]
Great! This fix worked for me thank you

**Comment by olufemi424** [comment]
```
rm -rf node_modules
npm install -g npm@latest
npm i core-util-is
```

Works ✔

**Comment by junhwong** [comment]
not works   https://github.com/nodejs/help/issues/2644#issuecomment-649075086

**Comment by Alsaheem** [comment]
thanks

**Comment by Base29** [comment]
Hello guys I am also getting this error on my node application when I build it for production 
`10|graphql |     at Module.<anonymous> (/root/blonk-api-v2/app.js:1:3572)
10|graphql |     at t (/root/blonk-api-v2/app.js:1:110)
10|graphql |     at /root/blonk-api-v2/app.js:1:903
10|graphql |     at Object.<anonymous> (/root/blonk-api-v2/app.js:1:913)
10|graphql | Error: Cannot find module '@babel/runtime/regenerator'
10|graphql |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
10|graphql |     at Function.Module._load (internal/modules/cjs/loader.js:562:25)
10|graphql |     at Module.require (internal/modules/cjs/loader.js:692:17)
10|graphql |     at require (internal/modules/cjs/helpers.js:25:18)
10|graphql |     at Object.<anonymous> (/root/blonk-api-v2/app.js:1:939)
10|graphql |     at t (/root/blonk-api-v2/app.js:1:110)
10|graphql |     at Module.<anonymous> (/root/blonk-api-v2/app.js:1:3572)
10|graphql |     at t (/root/blonk-api-v2/app.js:1:110)
10|graphql |     at /root/blonk-api-v2/app.js:1:903
10|graphql |     at Object.<anonymous> (/root/blonk-api-v2/app.js:1:913)`

This is my package.json
`{
    "name": "graphql-api",
    "version": "2.0.0",
    "description": "New API using GraphQL",
    "main": "app.js",
    "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1",
        "start": "nodemon --exec babel-node app.js",
        "build-dev": "rm -rf dist && webpack --mode development",
        "build-prod": "rm -rf dist && webpack --mode production"
    },
    "author": "Faisal Hussain",
    "license": "ISC",
    "dependencies": {
        "@babel/runtime": "^7.10.4",
        "@hapi/joi": "^15.1.1",
        "bcryptjs": "^2.4.3",
        "body-parser": "^1.19.0",
        "cors": "^2.8.5",
        "dotenv": "^8.2.0",
        "express": "^4.17.1",
        "express-graphql": "^0.9.0",
        "graphql": "^15.1.0",
        "graphql-merge-resolvers": "^1.1.16",
        "handlebars": "^4.7.6",
        "jsonwebtoken": "^8.5.1",
        "mongoose": "^5.9.18",
        "mongoose-autopopulate": "^0.12.2",
        "multer": "^1.4.2",
        "nodemailer": "^6.4.10",
        "nodemailer-mailgun-transport": "^2.0.0",
        "regenerator-runtime": "^0.13.5",
        "webpack": "^4.43.0",
        "webpack-cli": "^3.3.12",
        "webpack-node-externals": "^1.7.2"
    },
    "devDependencies": {
        "@babel/cli": "^7.10.3",
        "@babel/core": "^7.10.3",
        "@babel/node": "^7.10.3",
        "@babel/plugin-transform-async-to-generator": "^7.10.1",
        "@babel/plugin-transform-runtime": "^7.10.3",
        "@babel/polyfill": "^7.10.1",
        "@babel/preset-env": "^7.10.3",
        "babel-loader": "^8.1.0",
        "nodemon": "^2.0.4"
    }
}
`
I have nvm installed. I tried with node version 8.17.0 and 10.19.0.
NPM Version = 6.14.5

**Comment by Jay-Jay13** [comment]
Hello! Can someone help me? I am getting an error message.

Error: Cannot find module 'ejs'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at new View (/workspace/Test/node_modules/express/lib/view.js:81:14)
    at Function.render (/workspace/Test/node_modules/express/lib/application.js:570:12)
    at ServerResponse.render (/workspace/Test/node_modules/express/lib/response.js:1012:7)
    at /workspace/Test/MoreExpress/EJSDemo/app.js:9:6
    at Layer.handle [as handle_request] (/workspace/Test/node_modules/express/lib/router/layer.js:95:5)
    at next (/workspace/Test/node_modules/express/lib/router/route.js:137:13)

Here is the package.json:
{
  "name": "ejsdemo",
  "version": "3.0.1",
  "description": "Sample EJS exercise",
  "main": "app.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "license": "ISC",
  "dependencies": {
    "ejs": "^3.1.3"
  }
}

TIA.

**Comment by neokkers** [comment]
I had same issue. I just have renamed folder, which had been containing problem file, and it's working now. Maybe it was just reserved name or another nodejs package had it already

**Comment by Jay-Jay13** [comment]
All is good neokkers. I figured out a resolution already. Thanks anyway.

**Comment by asmaaomar66** [comment]
> Struggled with this error all day using a MERN stack. I fixed it by adding this line to my package.json in on the server side.
> 
> "client": "npm start --prefix client"

I can't found the server side in my package.json !! 
my package.json is 
{
  "name": "functions",
  "description": "Cloud Functions for Firebase",
  "scripts": {
    "lint": "eslint .",
    "serve": "firebase emulators:start --only functions",
    "shell": "firebase functions:shell",
    "start": "npm run shell",
    "deploy": "firebase deploy --only functions",
    "logs": "firebase functions:log"
  },
  "engines": {
    "node": "8"
  },
  "dependencies": {
    "algoliasearch": "^4.3.0",
    "firebase-admin": "^8.10.0",
    "firebase-functions": "^3.6.1"
  },
  "devDependencies": {
    "eslint": "^5.12.0",
    "eslint-plugin-promise": "^4.0.1",
    "firebase-functions-test": "^0.2.0"
  },
  "private": true
}

**Comment by J-Colin** [comment]
> > npm run clean && npm run build worked for me !!
> 
> after many tries, this was the only one that worked for me Thanks!!
> what i did:
> npm install clean
> npm clean
> npm build

This one worked for me. Thanks!

**Comment by TomaIvanovTomov** [comment]
Nodejs, npm, the biggest bullshits ever made. Every single command in terminal throws some error. Wasting days to resolve some errors before even start making the real app. How is that possible ....

**Comment by faizantariq1** [comment]
@TomaIvanovTomov  same happens to me when I start react-native. I just resolve the issue one by one and then good to go.

**Comment by Ezra-Siton-UIX** [comment]
I so much hate not semantic "no clue" how to solve `console errors` (0 from 10 the score of this console Nightmare Error ==> Hundreds of messages her).

**Comment by AndersonRioba** [comment]
> > npm run clean && npm run build worked for me !!
> 
> after many tries, this was the only one that worked for me Thanks!!
> what i did:
> npm install clean
> npm clean
> npm build

Thanks a lot for this.For my case i have used npm rebuild

**Comment by morange** [comment]
Hey folks,
i have an similar error on my mac (Catalina) :(

OS: macOS 10.15.6
node v10.18.0
npm 6.13.4

the error:
Error: Cannot find module './lib/js-yaml.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/foundation-cli/node_modules/js-yaml/index.js:4:12)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)


Who can give me a tip on how to eliminate the error?

Many thanks for your help
m.orange

**Comment by addaleax** [comment]
@morange That’s a different issue, so maybe you could open a new ticket here in the repository?

Either way, it might help to run `npm uninstall -g foundation-cli && npm install -g foundation-cli`.

**Comment by wcdeich4** [comment]
it finally worked for me after I did   sudo npm i cjs-loader (and make sure to install express, not just express-http-proxy)

**Comment by wcdeich4** [comment]
it finally worked for me after I did   sudo npm i cjs-loader (and make sure to install express, not just express-http-proxy)

**Comment by amrony** [comment]
![image](https://user-images.githubusercontent.com/36276984/93738496-bb907a80-fc07-11ea-8271-edf2ff1a1e05.png)

**Comment by mounika-13** [comment]
Error: Cannot find module 'C:\Users\CHEERA\Desktop\Web development using vscode\day9-10\app.js'
←[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)←[39m
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:841:27)←[39m
←[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)←[39m
←[90m    at internal/main/run_main_module.js:17:47←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: []
}



Can anyone solve this please

**Comment by saurishkar** [comment]
**I was trying to run `gulp -v` and i received this error:**
`
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'gulp-cli'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/usr/share/nodejs/gulp/bin/gulp.js:3:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
`

**Cause:** File permission issue. I ran the same with sudo permissions and it ran successfully.

Can you check if the directories that are being accessed for the modules have relevant permissions ?

**Comment by eliakimgarcia** [comment]
> @ subhrajitroy1989
> Probar
> `rm -rf node_modules`
> `npm install -g npm@latest`
> `npm i core-util-is`

excellent!, great!
I just want to know what these commands do

**Comment by PoojaDurgad** [comment]
@eliakimgarcia - 

>rm -rf node_modules 

this command can help you to remove node_modules at the root path
>npm install -g npm@latest

this command will help you to install the latest stable version of npm globally.
>npm i core-util-is

this command can help you to install core-util-is module

**Comment by Satyam62** [comment]
> @subhrajitroy1989
> Try
> `rm -rf node_modules`
> `npm install -g npm@latest`
> `npm i core-util-is`

thanks finally run my loader.js 
329 error
It is more benificial

**Comment by oshanz** [comment]
in my container, node_modules folder is located outside from the project directory. is there are any ENV variable that I can set the path. below setup doesn't work
```
:/app$ echo $PATH
/opt/node_modules/.bin:/usr/local/bundle/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by FizaWaseem** [comment]
What should I do? i was trying to create react native but this happen
PS F:\React-native> expo init AwesomeProject
internal/modules/cjs/loader.js:960
  throw err;
  ^

Error: Cannot find module 'C:\Users\FizaWaseem\AppData\Roaming\npm\node_modules\expo-cli\bin\expo.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

**Comment by shivamrawat262626** [comment]
how to resolve this error?

**Comment by shivamrawat262626** [comment]
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\bin\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\bin\node_modules\npm\bin\npm-cli.js'
←[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)←[39m
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:725:27)←[39m
←[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)←[39m
←[90m    at internal/main/run_main_module.js:17:47←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: []
}

**Comment by tarekahf** [comment]
> @subhrajitroy1989
> Try
> `rm -rf node_modules`
> `npm install -g npm@latest`
> `npm i core-util-is`

Following the above steps worked for me.

**Comment by vashisth00** [comment]
`[nodemon] 1.19.1
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node vb.js`
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'express'
Require stack:
- /Users/omkar.vilankar/woocommerce-react/vb.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/omkar.vilankar/woocommerce-react/vb.js:2:17)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/omkar.vilankar/woocommerce-react/vb.js' ]
}`

**Comment by cristianbb98** [comment]
If you don't have the solution, try changing imports.

For example:

wrong : import .... from **'src/'**
correct: import .... from **'../'**

It worked for me.

**Comment by amroeldewiny** [comment]
FIXED just reinstall nodeJS

**Comment by ankithans** [comment]
> @subhrajitroy1989
> Try
> `rm -rf node_modules`
> `npm install -g npm@latest`
> `npm i core-util-is`

worked 🙃
Thanks

**Comment by darkii321** [comment]
/root/.pm2/logs/server-error.log last 15 lines:
2|server   |     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
2|server   |     at Object.<anonymous> (/usr/share/pm2/lib/ProcessContainerFork.js:33:23)
2|server   |     at Module._compile (internal/modules/cjs/loader.js:778:30)
2|server   |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
2|server   |     at Module.load (internal/modules/cjs/loader.js:653:32)
2|server   |     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
2|server   |     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
2|server   |     at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
2|server   |     at startup (internal/bootstrap/node.js:283:19)
2|server   |     at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
2|server   |   message: 'Table \'panda2.service_category\' doesn\'t exist',
2|server   |   code: 'ER_NO_SUCH_TABLE',
2|server   |   errno: 1146,
2|server   |   sqlState: '42S02',

----------
some solution to that

**Comment by darde** [comment]
Ensure you have node and npm correctly installed:

type 
```bash
node -v

npm -v
```
if you're using some node version manager like *nvm* or *asdf* try changing your node version. This could be help.

**Comment by 3009aksingh** [comment]
Please help I am getting this error

nodemon] starting `node src/app.js`
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module '../../explain'
Require stack:
- G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\node_modules\mongodb\lib\core\wireprotocol\write_command.js
- G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\node_modules\mongodb\lib\core\wireprotocol\index.js
- G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\node_modules\mongodb\lib\core\topologies\server.js
- G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\node_modules\mongodb\lib\core\index.js
- G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\node_modules\mongodb\index.js
- G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\src\app.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (G:\PROJECTS\SSTECHSOLUTIONS WEBSITE\nodedynamic\node_modules\mongodb\lib\core\wireprotocol\write_command.js:7:17)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'G:\\PROJECTS\\SSTECHSOLUTIONS WEBSITE\\nodedynamic\\node_modules\\mongodb\\lib\\core\\wireprotocol\\write_command.js',
    'G:\\PROJECTS\\SSTECHSOLUTIONS WEBSITE\\nodedynamic\\node_modules\\mongodb\\lib\\core\\wireprotocol\\index.js',
    'G:\\PROJECTS\\SSTECHSOLUTIONS WEBSITE\\nodedynamic\\node_modules\\mongodb\\lib\\core\\topologies\\server.js',
    'G:\\PROJECTS\\SSTECHSOLUTIONS WEBSITE\\nodedynamic\\node_modules\\mongodb\\lib\\core\\index.js',
    'G:\\PROJECTS\\SSTECHSOLUTIONS WEBSITE\\nodedynamic\\node_modules\\mongodb\\index.js',
    'G:\\PROJECTS\\SSTECHSOLUTIONS WEBSITE\\nodedynamic\\src\\app.js'
  ]
}
[nodemon] app crashed - waiting for file changes before starting...

**Comment by HelyeFab** [comment]
> @subhrajitroy1989
> Try
> `rm -rf node_modules`
> `npm install -g npm@latest`
> `npm i core-util-is`

This worked for me too.
MacOS 11.5.1 
node v14.5.0
npm 7.20.3

**Comment by LokiInside** [comment]
SOLUTION `yarn upgrade --latest`

**Comment by Dlunadev7** [comment]
> > @subhrajitroy1989
> > `npm i core-util-is`
> 
> doing this only worked for me.

It worked for me, I'm building a server in apollo and that error was pissing me off already(?) hahaha.

**Comment by tahaluindo** [comment]
ok

**Comment by plexor** [comment]
> Had the same problem under macOS Catalina. A reinstall of the latest Node LTS version from (https://nodejs.org/de/download/) solved my problem.

Same here, I had the same issue with a cloned project, updating to the latest version of node solved the issue.

**Comment by UtkarshK95** [comment]
Try updating your dependencies in package.json and then update your node modules.
1. npm update (will update all dependencies, if you don't want that, try updating manually)
2. npm i
To check outdated dependencies use "npm outdated"

**Comment by sandeeprana19** [comment]
SR@DESKTOP-ON541FN MINGW64 /d/udemy/courses/react-complete-guide/advanced-handling-side-effects-using-reducers-&-using-the-context-API (main)
$ npm start

> react-complete-guide@0.1.0 start
> react-scripts start

'-using-the-context-API\node_modules\.bin\' is not recognized as an internal or external command,
operable program or batch file.
node:internal/modules/cjs/loader:1050
  throw err;
  ^

Error: Cannot find module 'D:\udemy\courses\react-complete-guide\react-scripts\bin\react-scripts.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1047:15)
    at Module._load (node:internal/modules/cjs/loader:893:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.14.0

**Comment by sandeeprana19** [comment]
I getting the above when I running npm start

**Comment by plexor** [comment]
did you ever get it ?

**Comment by Sonam-Sam** [comment]
node:internal/modules/cjs/loader:1051
  throw err;
  ^

Error: Cannot find module 'C:\Users\Sonam'
Require stack:
- internal/preload
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at internalRequire (node:internal/modules/cjs/loader:166:19)
    at Module._preloadModules (node:internal/modules/cjs/loader:1420:5)
    at loadPreloadModules (node:internal/process/pre_execution:705:5)
    at setupUserModules (node:internal/process/pre_execution:170:3)
    at prepareExecution (node:internal/process/pre_execution:132:5)
    at prepareMainThreadExecution (node:internal/process/pre_execution:55:3)
    at node:internal/main/run_main_module:10:1 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'internal/preload' ]
}

Node.js v20.9.0

I am getting this types of error when i tries to run my nestjs project. Can anyone help me resolve this?

**Comment by SHASHI-17** [comment]
> @subhrajitroy1989 Try `rm -rf node_modules` `npm install -g npm@latest` `npm i core-util-is`

thanks alot
